TSL: Introduce Chromatic Aberration

pashaydev opened the pull request:
**Description**

This pull request adds a new post-processing node for chromatic aberration effect in Three.js WebGPU/TSL.

![webgpu_postprocessing_ca](https://github.com/user-attachments/assets/6d54e57b-c206-4072-9f60-4c77b05965a3)

## Implementation Details

The `ChromaticAberrationNode` provides:
- **Strength control**: Adjustable intensity of the color separation effect (0-3 range)
- **Center point**: Customizable origin point for the radial distortion (defaults to screen center)
- **Scale factor**: Controls the stepped scaling of color channels from the center
- **Per-channel processing**: 
  - Red channel is scaled outward
  - Green channel remains at original scale  
  - Blue channel is scaled inward


## Example Usage

```javascript
import { chromaticAberration } from 'three/addons/tsl/display/ChromaticAberrationNode.js';

// Apply chromatic aberration to render output

// With static values (using uniform nodes)
const caPassStatic = chromaticAberration( outputPass, 1.0, null, 1.1 );

// With procedural nodes
const animatedStrength = sin( time ).mul( 0.5 ).add( 1.0 );
const dynamicCenter = vec2( cos( time ).mul( 0.1 ).add( 0.5 ), 0.5 );
const caPassProcedural = chromaticAberration( outputPass, animatedStrength, dynamicCenter, 1.0 );

postProcessing.outputNode = caPass;
```

## Demo
https://github.com/pashaydev/three.js/blob/ca_node/examples/webgpu_postprocessing_ca.html


PS:
I was looking into that effect build-in TLS but not found, if it will be helpful to someone like me i will be glad)

PashaDev2 committed:
Chromatic aberration TSL node

Example page for ca node

PashaDev2 committed:
Fix comma

PashaDev2 committed:
Remove unused delta from render loop in ca example

sunag commented on examples/jsm/tsl/display/CANode.js:
```diff
@@ -0,0 +1,310 @@
+ * @augments TempNode
+ * @three_import import { chromaticAberration } from 'three/addons/tsl/display/ChromaticAberrationNode.js';
+ */
+class ChromaticAberrationNode extends TempNode {
```
How about renaming the file `CANode` to the class name `ChromaticAberrationNode`? I know it's a long name, but we usually follow this logic.

Mugen87 replied:
Or we rename the class to `CANode` (similar to `GTAONode`, `SMAANode` or `SSRNode`). No matter how we decide, module and class name should be in sync.

sunag replied:
I agree, I would just vote for `ChromaticAberrationNode` as it seems like a more common name.

Mugen87 replied:
I'm okay with that!

PashaDev2 committed:
Rename CANode.js to ChromaticAberrationNode.js

Add nodeObject in ca params, for ability to use a procedural node

Update example gui to toggle procedural / static

PashaDev2 committed:
Fix the gui in example

sunag committed:
cleanup code style

sunag committed:
use nodeObject()

sunag committed:
Update webgpu_postprocessing_ca.html

sunag committed:
cleanup

sunag committed:
Merge branch 'dev' into pr/31236
